### PR TITLE
Release notes for 2.14

### PR DESCRIPTION
**Pull request by peternied**
### Description
Release notes for 2.14

### Issues Resolved
- Related https://github.com/opensearch-project/security/issues/4159

### Check List
- [ ] ~New functionality includes testing~
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-security.release-notes-2.14.0.0.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Compatible with OpenSearch 2.14.0
+
+### Enhancements
+* Check for and perform upgrades on security configurations ([#4251](https://github.com/opensearch-project/security/pull/4251))
+* Replace bouncy castle blake2b ([#4284](https://github.com/opensearch-project/security/pull/4284))
+* Adds saml auth header to differentiate saml requests and prevents auto login as anonymous user when basic authentication fails ([#4228](https://github.com/opensearch-project/security/pull/4228))
+* Dynamic sign in options ([#4137](https://github.com/opensearch-project/security/pull/4137))
+* Add index permissions for query insights exporters ([#4231](https://github.com/opensearch-project/security/pull/4231))
+* Add new stop words system index ([#4181](https://github.com/opensearch-project/security/pull/4181))
```
Comment (cwperks) [inline release-notes/opensearch-security.release-notes-2.14.0.0.md:11]: This was included in 2.13: https://github.com/opensearch-project/security/pull/4182

File: release-notes/opensearch-security.release-notes-2.14.0.0.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+* Add query assistant role and new ml system indices ([#4142](https://github.com/opensearch-project/security/pull/4142))
```
Comment (cwperks) [inline release-notes/opensearch-security.release-notes-2.14.0.0.md:12]: This was also included in 2.13: https://github.com/opensearch-project/security/pull/4143

File: release-notes/opensearch-security.release-notes-2.14.0.0.md (hunk 1 above)
Comment (peternied) [reply release-notes/opensearch-security.release-notes-2.14.0.0.md:11]: It wasn't included in the 2.13 release notes, keeping it here

**Review by reta — APPROVED** [review]

**Comment by willyborankin** [comment]
I think that we should mention that is `Improve cluster initialization reliability` is in the preview. Wdyt?

**Review by willyborankin — APPROVED** [review]

**Comment by willyborankin** [comment]
@cwperks upgrade for opensaml 4.3.2 has not been backported to 2.x

**Comment by cwperks** [comment]
@peternied I was unable to push to this branch so I opened a separate PR to address @willyborankin's comments: https://github.com/opensearch-project/security/pull/4307

**Comment by cwperks** [comment]
Closing this PR as https://github.com/opensearch-project/security/pull/4307 was merged.

**Pull request closed without merging**